library/data/component/src/main/java/org/quiltmc/qsl/component/impl/sync/codec/NetworkCodec.java

```diff
@@ -0,0 +1,122 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.component.impl.sync.codec;
+
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.entity.Entity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.network.PacketByteBuf;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.collection.DefaultedList;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.world.chunk.Chunk;
+import org.jetbrains.annotations.NotNull;
+import org.quiltmc.qsl.component.api.ComponentType;
+import org.quiltmc.qsl.component.api.Components;
+import org.quiltmc.qsl.component.impl.client.sync.ClientSyncHandler;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+public record NetworkCodec<T>(@NotNull BiConsumer<PacketByteBuf, T> encoder,
+							  @NotNull Function<PacketByteBuf, T> decoder) {
+	public static final NetworkCodec<BlockPos> BLOCK_POS = new NetworkCodec<>(
+			PacketByteBuf::writeBlockPos, PacketByteBuf::readBlockPos
+	);
+	public static final NetworkCodec<Integer> INT = new NetworkCodec<>(
+			PacketByteBuf::writeInt, PacketByteBuf::readInt
+	);
+	public static final NetworkCodec<ChunkPos> CHUNK_POS = new NetworkCodec<>(
+			PacketByteBuf::writeChunkPos, PacketByteBuf::readChunkPos
+	);
+	public static final NetworkCodec<Float> FLOAT = new NetworkCodec<>(
+			PacketByteBuf::writeFloat, PacketByteBuf::readFloat
+	);
+	// TODO: Handle the cases where MinecraftClient.getInstance().world is null
+	public static final NetworkCodec<BlockEntity> BLOCK_ENTITY = new NetworkCodec<>(
+			(buf, blockEntity) -> buf.writeBlockPos(blockEntity.getPos()),
+			buf -> MinecraftClient.getInstance().world.getBlockEntity(buf.readBlockPos())
+	);
+	public static final NetworkCodec<Entity> ENTITY = new NetworkCodec<>(
+			(buf, entity) -> buf.writeInt(entity.getId()),
+			buf -> MinecraftClient.getInstance().world.getEntityById(buf.readInt())
+	);
+	public static final NetworkCodec<Chunk> CHUNK = new NetworkCodec<>(
+			(buf, chunk) -> buf.writeChunkPos(chunk.getPos()),
+			buf -> CHUNK_POS.decode(buf)
+					.map(chunkPos -> MinecraftClient.getInstance().world.getChunk(chunkPos.x, chunkPos.z))
+					.orElseThrow()
+	);
+	public static final NetworkCodec<ItemStack> ITEMSTACK = new NetworkCodec<>(
```

Comment:
```suggestion
	public static final NetworkCodec<ItemStack> ITEM_STACK = new NetworkCodec<>(
```
Correct conversion between class name (`PascalCase`) and `static final` field name (`SCREAMING_SNAKE_CASE`).